### PR TITLE
Granular tests for interaction with UCAS website.

### PR DESCRIPTION
**Pull request by timabell**
The new integration tests in this patch test the interaction with the
live UCAS website. These tests are necessary as the generation of urls
for courses is complex and fragile to change on the UCAS side. These
tests will act as a canary for any changes to the UCAS site that we need
to react to and give an idea of what kind of change / failure we might
be dealing with.

The tests that are removed in this patch were too intertwined with
database, controllers etc. to give a precise view of the cause of any
failures.

Full integration of database + controllers + ucas + ui can be tested
with webdriver at the UI layer, ask Sau for his magic script :-)

The magic strings are now compiled into the library to give greater
certainty as to the behaviour of interaction with UCAS across all
environments. We can still respond quickly to changes as we are
sufficiently agile to be able to push changes to production rapidly.

* Refactored the UCAS-handling code out of the controller for cleaner
  testing.

### CODE REVIEW

**Comment by the-daniel-rothig** [comment]
nb merged build config fix

**Comment by defong** [comment]
Magic script is here, it uses linqpad, very self-contained, windows + selenium + chrome based.

http://share.linqpad.net/46m6ml.linq

File: tests/Integration.Tests/UcasLink/UcasUrlBuilderIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+using GovUk.Education.SearchAndCompare.Api.Ucas;
+using NUnit.Framework;
+
+namespace GovUk.Education.SearchAndCompare.Api.Integration.Tests.UcasLink
+{
+    /// <summary>
+    /// Set of tests for interacting with the live UCAS website,
+    /// in increasing order of pedanticness.
+    /// </summary>
+    [TestFixture]
```
Comment (defong) [inline tests/Integration.Tests/UcasLink/UcasUrlBuilderIntegrationTests.cs]: -    [Category("Integration")]
-    [Category("Integration_Ucas")]
-    [Explicit]

Can you add that so we can don't needless run the unit tests

**Comment by timabell** [comment]
Rebased to remove odd duplicate commit in branch history.

**Pull request merged**